### PR TITLE
Fix path issues

### PR DESCRIPTION
**Pull request by frarees**
- Bundling from a folder no longer prepends '/'
- Building from files no longer prepends './'

### CODE REVIEW

**Comment by avivbeeri** [comment]
I am fine with this change, but it depends on this DOME PR being up to scratch: https://github.com/avivbeeri/dome/pull/62

**Comment by avivbeeri** [comment]
I'm having issues with this as is, because it trims off the first character of paths beginning with a directory, which is impeding my tests of the DOME PR changes.

**Comment by frarees** [comment]
Can you share how you're executing nest, and the output you get? (nest output, and possibly `tar -tvf <egg>`)

I got an expected output with:
`~/Documents/nest/nest -z -o game.egg -- <folder>`
`~/Documents/nest/nest -z -o game.egg <folder>/*`
`~/Documents/nest/nest -z -o game.egg <file>`

**Comment by avivbeeri** [comment]
In the DOME /examples/spaceshooter folder, execute `nest -zo game.egg -- *.wren res`, and when you run it with DOME, it results in `Error: There was a problem reading res/ship1.png from the bundle: null record
Runtime error: Could not find file: ./res/ship1.png`

When you use `tar` to extract the files...
Screen Shot 2020-04-16 at 14 33 55
Note the `es/` on all the paths here.

**Comment by frarees** [comment]
I see, I can indeed reproduce this. Sharing the whole output for the record:

```
frarees@MBP spaceshooter % ~/Documents/nest/nest -zo game.egg -- *.wren res
2
Bundling: main.wren
Bundling: res/music.wav
Bundling: res/enemy.png
Bundling: res/Hit_Hurt.wav
Bundling: res/sfx.bfxrlibrary
Bundling: res/heart-empty.png
Bundling: res/Laser_Shoot.wav
Bundling: res/heart-full.png
Bundling: res/ship1.png
Bundling: res/ship2.png
Bundling: res/Explosion.wav
Bundling: res/around-the-corner.ogg
Bundling: res/AerisPiano.ogg
Created bundle game.egg.
frarees@MBP spaceshooter % tar -tvf game.egg                               
-rw-rw-r--  0 0      0        8415 Jan  1  1970 main.wren
-rw-rw-r--  0 0      0     2408718 Jan  1  1970 es/music.wav
-rw-rw-r--  0 0      0         188 Jan  1  1970 es/enemy.png
-rw-rw-r--  0 0      0       20600 Jan  1  1970 es/Hit_Hurt.wav
-rw-rw-r--  0 0      0         415 Jan  1  1970 es/sfx.bfxrlibrary
-rw-rw-r--  0 0      0         166 Jan  1  1970 es/heart-empty.png
-rw-rw-r--  0 0      0       55224 Jan  1  1970 es/Laser_Shoot.wav
-rw-rw-r--  0 0      0         166 Jan  1  1970 es/heart-full.png
-rw-rw-r--  0 0      0         183 Jan  1  1970 es/ship1.png
-rw-rw-r--  0 0      0         182 Jan  1  1970 es/ship2.png
-rw-rw-r--  0 0      0       56860 Jan  1  1970 es/Explosion.wav
-rw-rw-r--  0 0      0     1936069 Jan  1  1970 es/around-the-corner.ogg
-rw-rw-r--  0 0      0     5709331 Jan  1  1970 es/AerisPiano.ogg
```

File: src/main.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ void listDir(mtar_t* tar, char* directory, char* subtract) {
       if ((strcmp(de->d_name, ".") != 0) && (strcmp(de->d_name, "..") != 0)) {
         char path[PATH_MAX];
         snprintf(path, PATH_MAX, "%s/%s", directory, de->d_name);
-        char* finalFileName = path + strlen(subtract);
+        char* finalFileName = path + strlen(subtract) + 1;
```
Comment (avivbeeri) [inline src/main.c]: What's the purpose of this `+ 1`? Removing it resolves my trimming issue, but it may have additional effects.

Comment (frarees) [reply src/main.c]: I think it exists cause we use `subtract`. I'm going to share some changes where I cleaned up that part. So far everything is working, except for one thing: when passing folders, it won't account for the extra trailing slash. I don't know if there's a C standard library or POSIX way to handle this properly, but if there is, I'd rather use that than manually searching for trailing slash.

Comment (avivbeeri) [reply src/main.c]: Unfortunately there isn't a standard or POSIX way to do that as far as I know. I think we have to assume that the end of the path string contains any number of trailing slashes and shorten the path accordingly.

**Comment by frarees** [comment]
> any number of trailing slashes

I went for just the last trailing slash. This is similar to what dome does currently on its io.c

**Comment by avivbeeri** [comment]
I think that will be sufficient for now. It would be uncommon to see paths with multiple slashes. I need to do some final tests, but otherwise this looks good.

**Comment by frarees** [comment]
I discovered an issue with nest where it won't behave as it did: passing a folder will get the full path to it, instead of _basing_ it. I will work on this, and then get back to dome.

**Comment by frarees** [comment]
I _think_ I've got every case covered, including subfolder(s), just one folder, trailing slash, ...
Give it a go, and let me know if you find something else. Otherwise, I can rebase/squash everything into one cleaned up commit.

**Comment by avivbeeri** [comment]
This seems to work for all cases I can think of. Squash and we can merge :)

**Pull request merged**